src/activities/morse_code/MorseCode.qml

```diff
@@ -0,0 +1,567 @@
+/* GCompris - MorseCode.qml
+ *
+ * Copyright (C) 2016 SOURADEEP BARUA <sourad97@gmail.com>
+ *
+ * Authors:
+ *   SOURADEEP BARUA <sourad97@gmail.com>
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+import QtQuick 2.1
+import "../../core"
+import GCompris 1.0
+import "dataset.js" as Dataset
+
+ActivityBase {
+    id: activity
+    property string url : "qrc:/gcompris/src/activities/morse_code/resource/"
+    onStart: focus = true
+    onStop: {}
+
+
+    pageComponent: Image {
+        id: background
+        source: url+"background.svg"
+        fillMode: Image.PreserveAspectCrop
+        sourceSize.width: Math.max(parent.width, parent.height)
+
+        signal start
+        signal stop
+
+        Component.onCompleted: {
+            activity.start.connect(start)
+            activity.stop.connect(stop)
+        }
+
+        // Add here the QML items you need to access in javascript
+        QtObject {
+            id: items
+            property Item main: activity.main
+            property alias background: background
+            property alias bar: bar
+            property alias bonus: bonus
+            property alias score: score
+            property alias textInput: textInput
+            property var dataset: Dataset.get()
+            property bool toAlpha: dataset[currentLevel].toAlpha
+            property string questionText: dataset[currentLevel].question
+            property string questionValue
+            property string instruction: dataset[currentLevel].instruction
+            property int currentLevel: 0
+            property int numberOfLevel: dataset.length
+
+
+            onQuestionValueChanged: {
+                textInput.text = ''
+
+                morseConverter.alpha = 0
+                if(toAlpha)
```

Comment:
if I'm not wrong you update the keyboard at each question. You should only do it when you change level as it's always the same mode for all the questions in a same level.
Populating the keyboard is quite consuming (it takes 4-5 seconds on my phone) so it would be better to not do it too often ;).